app/controllers/api/v1/book_controller.rb

```diff
@@ -2,7 +2,20 @@ module Api
   module V1
     class BookController < ApiController
       def index
-        render_paginated Book.all
+        if params[:isbn].nil?
+          render_paginated Book.all
+        else
+          show_isbn
+        end
+      end
+
+      def show_isbn
+        response = OpenLibraryService.book_info(params[:isbn])
```

Comment:
1. Use `require` and rescue the error to valid the request params `isbn`
2. Encapsule you creation of ibsn in a private method
3. Use you initialize and call your services as this: `OpenLibraryService.new(isbn).book_info`